[GTK] Split hardware acceleration information in webkit://gpu

carlosgcampos opened the pull request:
#### 288d895da13b0de1885f518d3be47acd726fdffc
<pre>
[GTK] Split hardware acceleration information in webkit://gpu
<a href="https://bugs.webkit.org/show_bug.cgi?id=256035">https://bugs.webkit.org/show_bug.cgi?id=256035</a>

Reviewed by Žan Doberšek.

When using DMA-BUF for rendering we use a different EGL platform in the
UI and Web processes, which might have different extensions and
properties, but we are always using only the one used in the UI process.

* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::~PlatformDisplay): Handle the case of
destructor being called before the exit handlers.
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::uiProcessContextIsEGL):
(WebKit::WebKitProtocolHandler::handleGPU):

Canonical link: <a href="https://commits.webkit.org/263668@main">https://commits.webkit.org/263668@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/69d327de94fc1711634a33d7dc50f45041b2dce1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4786 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4906 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5072 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6297 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4914 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4778 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5064 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4876 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5153 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4867 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4947 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4285 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6310 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2435 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4280 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9251 "1 flakes 141 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4297 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4355 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5926 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4762 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3875 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4273 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8332 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/609 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4633 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

shivamidow commented on Source/WebCore/platform/graphics/PlatformDisplay.cpp:
```diff
@@ -221,11 +221,19 @@ void PlatformDisplay::sharedDisplayDidClose()
+
+PlatformDisplay::~PlatformDisplay()
+{
+    if (m_eglDisplay != EGL_NO_DISPLAY && eglDisplays().remove(this))
```
/* Unofficial Review */
No guard here? eglDIsplays() and terminateEGLDisplay() are in USE(EGL).

carlosgcampos committed:
[GTK] Split hardware acceleration information in webkit://gpu
https://bugs.webkit.org/show_bug.cgi?id=256035

Reviewed by Žan Doberšek.

When using DMA-BUF for rendering we use a different EGL platform in the
UI and Web processes, which might have different extensions and
properties, but we are always using only the one used in the UI process.

* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::~PlatformDisplay): Handle the case of
destructor being called before the exit handlers.
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::uiProcessContextIsEGL):
(WebKit::WebKitProtocolHandler::handleGPU):

Canonical link: https://commits.webkit.org/263668@main
